src/main/scala/com/databricks/spark/avro/ConfluentSparkAvroUtils.scala

```diff
@@ -60,12 +66,59 @@ class ConfluentSparkAvroUtils(schemaRegistryURLs: String) extends Serializable {
 
   @transient private lazy val deserializer = new KafkaAvroDeserializer(schemaRegistry)
 
+  @transient private lazy val AES_MAGIC_BYTES = Set[Byte](2, 3)
```

Comment:
Why is everything marked lazy? I suspect it is for avoiding initializing things that are sometimes not needed? If that's the case, maybe the class can be broken down to classes where in each one everything is needed?

Reply:
made constants not lazy, other things are lazy and transient to avoid being serialized which is not quite possible for example for AWSKMS client

Reply:
anything `@transient` wouldn't get serialized regardless of whether it's `lazy`, so feel free to remove `lazy` again. The two main reason for ever having `lazy`ies are: (1) avoid expensive initialization if there's a chance that the value is not going to be used, (2) create cyclic dependencies.

Reply:
then I'd prefer to keep this for AWSKMS as we don't always need it (we can read unencrypted topics with this same code) and it's expensive.

Reply:
Makes sense. Consider breaking this class in the future into several more focused classes. For example "AvroSerializers" and "EncryptedAvroSerializers" and maybe they share some common trait. "Utils" classes get hairy very fast since any random thing can be added to them and just marked lazy for performance. When that happens, it would be better to pass AWSKMS as constructor parameter so the same instance can be shared by different parts of the program (or a mock can be passed in tests)